src/equations/ideal_glm_mhd_2d.jl

```diff
@@ -279,6 +281,181 @@ end
     return f
 end
 
+"""
+    flux_nonconservative_powell2(u_ll, u_rr, orientation::Integer,
+                                equations::IdealGlmMhdEquations2D)
+
+Non-symmetric two-point flux discretizing the nonconservative (source) term of
+Powell and the Galilean nonconservative term associated with the GLM multiplier
+of the [`IdealGlmMhdEquations2D`](@ref).
+
+This implementation uses a non-conservative term that can be written as the product
+of local and symmetric parts. It is equivalent to the non-conservative flux of Bohm
+et al. (`flux_nonconservative_powell`) for conforming meshes but it yields different
+results on non-conforming meshes(!).
+## References
+- Rueda-Ramírez, Gassner (2023). A Flux-Differencing Formula for Split-Form Summation By Parts
+  Discretizations of Non-Conservative Systems. https://arxiv.org/pdf/2211.14009.pdf.
+"""
+@inline function flux_nonconservative_powell2(u_ll, u_rr, orientation::Integer,
```

Comment:
I added the `2` because the "new" flux is algebraically equivalent to the Powell flux **in conforming meshes**. In non-conforming meshes, this flux gives different results.

What would be a good name for this flux? `flux_nonconservative_powell_local_times_symmetric`? What do you think, @sloede?

Context: The new flux is a different way to write the Powell flux as the product of a local and a symmetric part: `flux(u_L, u_R) = local(u_L) * symmetric(u_L, u_R)`, where `symmetric(u_L, u_R) = symmetric(u_R, u_L)`.

Reply:
Good question. I am not sure about it, except that just adding a `2` is the worst of all options 😬. Is this a flux specifically to be used in non-conforming meshes, such that `flux_nonconservative_powell_nonconforming` would make sense?

Reply:
> Is this a flux specifically to be used in non-conforming meshes?

No, the flux was added because its form (local*symmetric) is needed for subcell limiting. It can be used in conforming or non-conforming meshes. 

In fact, there is preliminary evidence that suggests that this new flux could be more prone to instabilities than `flux_nonconservative_powell` in non-conforming meshes. This needs more investigation... For now, I just added a note in the docstring:
```
"""
    flux_nonconservative_powell2(u_ll, u_rr, orientation::Integer,
                                equations::IdealGlmMhdEquations2D)

Non-symmetric two-point flux discretizing the nonconservative (source) term of
Powell and the Galilean nonconservative term associated with the GLM multiplier
of the [`IdealGlmMhdEquations2D`](@ref).

This implementation uses a non-conservative term that can be written as the product
of local and symmetric parts. It is equivalent to the non-conservative flux of Bohm 
et al. (`flux_nonconservative_powell`) for conforming meshes but it yields different 
results on non-conforming meshes(!).
## References
- Rueda-Ramírez, Gassner (2023). A Flux-Differencing Formula for Split-Form Summation By Parts
  Discretizations of Non-Conservative Systems. https://arxiv.org/pdf/2211.14009.pdf.
"""
```

Reply:
I have the following candidates for the new name:
* `flux_nonconservative_powell_local_times_symmetric`
* `flux_nonconservative_powell_local_symmetric`
* `flux_nonconservative_powell_subcell` (although it is not only for subcell limiting...)
* `flux_nonconservative_powell2` :) 

What's your favorite?

Reply:
flux_nonconservative_powell_local_symmetric would be my favorite, although not by much. What about flux_nonconservative_powell_rueda_gassner?